finetune/comparison.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+
+from finetune.base import BaseModel, CLASSIFICATION, REGRESSION, SEQUENCE_LABELING
+from finetune.errors import InvalidTargetType
+import tensorflow as tf
+
+
+class Comparison(BaseModel):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _text_to_ids(self, X_1, X_2, max_length=None):
+        max_length = max_length or self.config.max_length
+        forward_pairs = self.encoder.encode_multi_input(X_1, X_2, max_length=max_length, verbose=self.config.verbose)
+        backward_pairs = self.encoder.encode_multi_input(X_2, X_1, max_length=max_length, verbose=self.config.verbose)
+        seq_array_fw = self._array_format(forward_pairs)
+        seq_array_bw = self._array_format(backward_pairs)
+        token_ids = np.stack([seq_array_fw.token_ids, seq_array_bw.token_ids], 1)
+        mask = np.stack([seq_array_fw.mask, seq_array_bw.mask], 1)
+
+        return token_ids, mask
+
+    def finetune(self, X_1, X_2, Y, batch_size=None):
+        """
+        :param Xs: An iterable of lists or array of text, shape [batch, n_inputs, tokens]
+        :param Y: integer or string-valued class labels. It is necessary for the items of Y to be sortable.
+        :param batch_size: integer number of examples per batch. When N_GPUS > 1, this number
+                           corresponds to the number of training examples provided to each GPU.
+        """
+        if self.target_type is None:
```

Comment:
I didnt know whether it was desired to support regression for this?

Reply:
Ill just assume this is all tided up in your PR #64  and will leave this for now.